### ISSUE TITLE
Custom loss functions (and ideally, task-specific)

### ISSUE THREAD

**Issue by blengerich** [cnellington/Contextualized#55]
Would be great to have customizable loss functions, and if possible, task-specific loss functions.

**Comment by cnellington** [comment]
Notes from last discussion: easiest implementation would be to split tasks from the dataloader into T_x and T_y and pass these to a generic loss function which handles T_x, T_y, and y_hat to route predictions to task-specific losses.
